File: frontend/src/lib/poolInputHealth.js
const SOURCES = { odds: 'SvS-odds', sharp: 'Pinnacle-odds', streck: 'folkets streck', none: 'underlag saknas' }

export function poolInputTitle(health) {
  if (!health || health.level === 'ok') return null
  if (health.missing_all > 0) return `Kompletta 1X2-odds saknas från båda källorna i ${health.missing_all}/${health.n_matches} matcher`
  if (health.missing_sharp > 0) {
    // pool-sharp-freshness-v1: ett cachat men inaktuellt pris räknas som saknat.
    const stale = health.stale_sharp > 0 ? ` (${health.stale_sharp} med inaktuellt pris)` : ''
    return `Pinnacle 1X2 saknas i ${health.missing_sharp}/${health.n_matches} matcher${stale}`
  }
  if (health.missing_svs > 0) return `SvS 1X2 saknas i ${health.missing_svs}/${health.n_matches} matcher`
  return `Ö/U-underlag saknas i ${health.missing_total}/${health.n_matches} matcher`
}

export function sourceLabel(source) { return SOURCES[source] || 'okänd källa' }

export function poolReviewText(health, scope = 'Aktuell analys') {
  return [
    'Granska saknat oddsunderlag i Spelkompisen.',
    `Spel: ${health.product} · omgång ${health.draw_number} · ${scope}`,
    `Analyssvar hämtat: ${health.analysis_fetched_at || 'okänd tid'}`,
    `Kontroll: ${health.version}${health.freshness_version ? ` + ${health.freshness_version}` : ''} · ${health.n_matches} matcher`,
    `Ofullständig SvS 1X2: ${health.missing_svs}. Pinnacle 1X2: ${health.missing_sharp}${health.stale_sharp ? ` (varav ${health.stale_sharp} med inaktuellt cachat pris)` : ''}. Båda: ${health.missing_all}. Ö/U: ${health.missing_total}.`,
    ...health.issues.map(m => `${m.event_number}. ${m.description}: saknar ${m.missing.join(', ')}. Sannolikhetsbas: ${sourceLabel(m.prob_source)}.${m.reason ? ` Orsak: ${m.reason}.` : ''}`),
    ...health.issues.filter(m => m.reserve_total).map(m => {
      const r = m.reserve_total
      return `${m.event_number}. Ö/U-reserv: ${r.label}; status ${r.status}; färsk/tillgänglig ${r.available ? 'ja' : 'nej'}; lina ${r.line ?? '–'}; Över ${r.over_odds ?? '–'} / Under ${r.under_odds ?? '–'}; observerat ${r.observed_at || 'okänt'}. Inte sharp, används inte av byggaren.`
    }),
    'Kontrollera källhälsa, matchning (inklusive pool_match_diagnostic), marknadsutbud och observationstider. Anta inte att allt beror på namn.',
    'Detta beskriver analysens tillgängliga priser. Pinnacle används bara om priset är högst 90 min gammalt och länken inte observerats tappad efter priset; SvS-oddsen är inte bevis på färskhet, och inget av det är en giltig historisk PIT-capture. Bakfyll inte odds eller ändra frysta kuponger.',
  ].join('\n')
}

File: backend/app/pool_input_health.py
"""Presentationsvarning för det underlag analysen/bygget faktiskt fick.

Inga nätanrop, modelländringar eller antaganden om varför priser saknas.
Ett befintligt pris är inte ett bevis på färsk presence.
"""
import math

VERSION = "pool-input-health-v1"
SIGNS = ("1", "X", "2")


def _price(value):
    return isinstance(value, (int, float)) and not isinstance(value, bool) and math.isfinite(value) and value > 1


def report(analysis):
    issues = []
    counts = {"missing_sharp": 0, "missing_svs": 0, "missing_all": 0, "missing_total": 0}
    for match in analysis.matches:
        svs = all(s in match.outcomes and _price(match.outcomes[s].odds) for s in SIGNS)
        sharp = all(s in match.outcomes and _price(match.outcomes[s].sharp_odds) for s in SIGNS)
        line = match.total_line
        total = (isinstance(line, (int, float)) and math.isfinite(line) and line > 0
                 and _price(match.total_over_odds) and _price(match.total_under_odds))
        missing = []
        for key, missing_value, label in (
                ("missing_svs", not svs, "SvS 1X2"),
                ("missing_sharp", not sharp, "Pinnacle 1X2"),
                ("missing_total", not total, "Pinnacle Ö/U")):
            if missing_value:
                counts[key] += 1
                missing.append(label)
        if not svs and not sharp:
            counts["missing_all"] += 1
        if missing:
            issues.append({"event_number": match.event_number,
                           "description": match.description or f"Match {match.event_number}",
                           "missing": missing, "no_complete_1x2": not svs and not sharp,
                           "prob_source": match.prob_source})
    return {"version": VERSION, "product": analysis.product, "draw_number": analysis.draw_number,
            "analysis_fetched_at": analysis.fetched_at, "n_matches": len(analysis.matches),
            "level": "error" if counts["missing_all"] else "warning" if issues else "ok",
            **counts, "issues": issues}
